src/coreclr/jit/importer.cpp

```diff
@@ -10622,6 +10674,16 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 op1 = impPopStack().val; // Ptr
                 assertImp(varTypeIsStruct(op2));
 
+                if (varTypeIsStruct(op2) && (op2->IsCall() || op2->OperIs(GT_RET_EXPR)))
+                {
+                    unsigned             callTmp = lvaGrabTemp(true DEBUGARG("spill struct call to tmp"));
+                    CORINFO_CLASS_HANDLE cls =
+                        op2->IsCall() ? op2->AsCall()->gtRetClsHnd : op2->AsRetExpr()->gtInlineCandidate->gtRetClsHnd;
+                    lvaSetStruct(callTmp, cls, false);
+                    impStoreToTemp(callTmp, op2, CHECK_SPILL_ALL);
+                    op2 = gtNewLclVarNode(callTmp);
+                }
```

Comment:
What was the example where my `impStoreStruct` change wasn't working? I'd hope we can avoid changes like these outside `impStoreStruct`.

Reply:
yep, I'll definitely clean it up once I get it to a green state 🙂